### DISCUSSION TITLE
Program problems of ESPI Library in stm32f103c8t6

### DISCUSSION THREAD

**Discussion by JingtaiChen** [Bodmer/TFT_eSPI#2015] [Q&A - General] [answered]
Hello, friend
My problem is that the ESPI library is applied to STM32 Series MCU. First, I did an experiment (Application of ESPI on esp32). This experiment perfectly displayed the content. But ask
The question appeared in the ESPI library and STM32. I modified the user_ Setup。 The pin definition of the. H file (other definitions I tested in esp32 are normal), as shown in the figure:
![image](https://user-images.githubusercontent.com/111559992/189541070-fce51b05-977b-4c7a-b796-3d860031e337.png)
But de reported this fault. As shown in the figure:
![image](https://user-images.githubusercontent.com/111559992/189541118-6c94076d-912e-4d64-a09c-427c31834ae1.png)
The above is only one of the questions. When I continue to experiment, I find that esp32 does not have port multiplexing function? However, when I repeatedly define data and clock on the SPI port used by the SD card
The display will not be available. However, this problem does not matter, because I can solve the problem by avoiding several ports used by the SD card. I look forward to the answer from the great God.

**Comment by Bodmer** [comment, marked as answer]
I do not get the same errors. Make sure you have the correct board package loaded into the Arduino IDE and not an old deprecated version. I have version 2.3.0:

![image](https://user-images.githubusercontent.com/15824805/189547433-981047b8-bc33-4c49-9733-ff2ef4058100.png)

**Comment by JingtaiChen** [comment]
Sorry, I'm not using the official stm32dino library, but a third-party library, because the third-party library is optimized for F1. I think the performance of this library is better than the official library.
![image](https://user-images.githubusercontent.com/111559992/189550169-2156e42b-91aa-4dc3-9a5f-0d55e58554de.png)

**Reply by Bodmer** [reply]
It appears the board package is therefore not compatible with the library. I am unable to help with that.

**Reply by JingtaiChen** [reply]
Well, thank you for your reply. I'm looking for other ways
